lib/spark/dsl/entity.ex

```diff
@@ -64,16 +64,16 @@ defmodule Spark.Dsl.Entity do
           recursive_as: atom,
           examples: [String.t()],
           entities: [t],
-          deprecations: [{atom, String.t()}],
+          deprecations: Keyword.t(String.t()),
           describe: String.t(),
           snippet: String.t(),
           args: [atom],
-          links: [{atom, [String.t()]}],
-          hide: boolean,
+          links: Keyword.t([String.t()]),
+          hide: [atom],
           modules: [String.t()],
           no_depend_modules: [atom],
-          schema: OptionsHelpers.schema(),
-          auto_set_fields: [{atom, any}],
+          schema: Keyword.t(OptionsHelpers.schema()),
```

Comment:
This should probably be the actual definition of OptionsHelpers.schema, and what is currently called `schema` should probably be called `option_schema` or something.